src/main/java/core/basesyntax/HelloWorld.java

```diff
@@ -1,9 +1,49 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ReaderService;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.WriterService;
+import core.basesyntax.serviceimpl.CsvToFruitTransactionConverter;
+import core.basesyntax.serviceimpl.ReaderServiceImpl;
+import core.basesyntax.serviceimpl.ReportGeneratorImpl;
+import core.basesyntax.serviceimpl.ShopServiceImpl;
+import core.basesyntax.serviceimpl.WriterServiceImpl;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.List;
+import java.util.Map;
+
 public class HelloWorld {
-    // HINT: In the `public static void main(String[] args)` it is better to create instances of your classes, 
-    // and call their methods, but do not write any business logic in the `main` method!
-}
+    public static void main(String[] arg) {
+        ReaderService<List<String>> readerService = new ReaderServiceImpl();
+        List<String> inputReport = readerService.readFromFile("src/main/java/resources/input.csv");
+
+        DataConverter<List<String>,
+                List<FruitTransaction>> dataConverter = new CsvToFruitTransactionConverter();
+        final List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        Map<FruitTransaction.Operation,
```

Comment:
Avoid using switch-cases and if-else constructions; use the Strategy pattern instead. Here, it's correctly applied by using a map of operation handlers.